src/productcatalogservice/server.go

```diff
@@ -164,6 +169,27 @@ func initStackDriverTracing() {
 	log.Warn("could not initialize stackdriver exporter after retrying, giving up")
 }
 
+// Initialize OTel trace provider that exports to Cloud Trace
+func initTraceProvider() {
+	// Initialize exporter OTel Trace -> Google Cloud Trace
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	exporter, err := texporter.NewExporter(texporter.WithProjectID(projectID))
```

Comment:
Make sure the projectId is not null or empty as well as check the presense of the GOOGLE_APPLICATION_CREDENTIALS env variable to make sure that auth will work.

Reply:
checking that projectID isn't empty, I don't think it's necessary to check credentials, would be the responsibility of the [exporter](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/4bf57be619cd46e7dfce46107fc7baa368ebc9ef/exporter/trace/README.md#authentication)